dashboard/app/controllers/datablock_storage_controller.rb

```diff
@@ -0,0 +1,347 @@
+  #   Library Manifest API (shared table metadata)         #
+  ##########################################################
+
+  def get_library_manifest
+    render json: DatablockStorageLibraryManifest.instance.library_manifest
+  end
+
+  def set_library_manifest
+    library_manifest = JSON.parse params[:library_manifest]
+    DatablockStorageLibraryManifest.instance.update!(library_manifest: library_manifest)
+    render json: true
+  end
+
+  ##########################################################
+  #   Project API                                          #
+  ##########################################################
+
+  # Returns true if there is any data for the project
+  def project_has_data
+    is_there_data = DatablockStorageTable.exists?(project_id: @project_id) || DatablockStorageKvp.exists?(project_id: @project_id)
+    render json: is_there_data
+  end
+
+  # Deletes all datablock storage data for the project
+  def clear_all_data
+    DatablockStorageTable.where(project_id: @project_id).delete_all
+    DatablockStorageKvp.where(project_id: @project_id).delete_all
+    DatablockStorageRecord.where(project_id: @project_id).delete_all
+    render json: true
+  end
+
+  ##########################################################
+  #   Project Use Datablock Storage API                    #
+  ##########################################################
+
+  # TODO: post-firebase-cleanup, remove this code: #56994
+  def use_datablock_storage
+    ProjectUseDatablockStorage.set_data_block_storage_for!(params[:channel_id], true)
+    render json: true
+  end
+
+  # TODO: post-firebase-cleanup, remove this code: #56994
+  def use_firebase_storage
+    ProjectUseDatablockStorage.set_data_block_storage_for!(params[:channel_id], false)
+    render json: true
+  end
+
+  ##########################################################
+  #   Private                                              #
+  ##########################################################
+
+  private def shared_table?
+    ActiveRecord::Type::Boolean.new.cast(params[:is_shared_table])
+  end
+
+  private def find_table_or_shared_table
+    shared_table? ?
+      DatablockStorageTable.find_shared_table(params[:table_name]) :
+      find_table
+  end
+
+  private def find_table
+    DatablockStorageTable.find([@project_id, params[:table_name]])
+  rescue ActiveRecord::RecordNotFound
+    raise StudentFacingError, "You tried to use a table called \"#{params[:table_name]}\" but that table doesn't exist in this app"
+  end
+
+  private def where_table
+    DatablockStorageTable.where(project_id: @project_id, table_name: params[:table_name])
+  end
+
+  private def table_or_create
+    where_table.first_or_create
+  end
+
+  private def validate_channel_id
+    project = Project.find_by_channel_id(params[:channel_id])
+    unless SUPPORTED_PROJECT_TYPES.include? project.project_type
+      raise "DatablockStorage is only available for applab and gamelab projects"
```

Comment:
For the future: I think these messages will eventually need to be translated

Reply:
resolving since Seth addressed this in a different comment